cromshell

```diff
@@ -649,127 +811,66 @@ function runSubCommandOnWorkflowId()
 # The bash equivalent to `if __name__ == '__main__':
 if ${ISCALLEDBYUSER} ; then
 
-  #Check given arguments:
-  if [[ $# -gt $MAXARGS ]] ; then
-    simpleUsage
-    exit 1
-  elif [[ $# -lt $MINARGS ]] ; then
-    simpleUsage
-    exit 2
-  fi
+  # Check the remaining arguments for help and display it if we have to: 
+  checkForComplexHelpArgument $@
 
-  #Read args:
-  while [ $# -gt 0 ] ; do
-
-    case "$1" in
-      -h|--h|--help|-help|help)
-        usage;
-        exit 0;
-        ;;
-      submit|status|logs|execution-status-count|metadata|slim-metadata|timing|abort|notify|list|fetch-all|fetch-logs|list-outputs) 
-        # Get our sub-command:
-        SUB_COMMAND=$1
-        shift
-        # We'll manually handle the rest of the options:
-        break
+  # Get top-level arguments to script itself:
+  while getopts "h" opt ; do
+    case ${opt} in
```

Comment:
This help checking seems redundant / strange considering the more advanced help check you added elsewhere.